### DISCUSSION TITLE
Creating a pack

### DISCUSSION THREAD

**Discussion by suhr** [GitoxideLabs/gitoxide#692] [Q&A] [answered]
I'm writing a tiny git server. It needs a way to construct a pack for sending over the network.
How to do it using gitoxide? https://docs.rs/git-pack/latest/git_pack/ doesn't seem to have functions for pack creation...

**Comment by Byron** [comment, marked as answer]
`gix free pack create -h` allows to build packs from the command-line, and [its implementation](https://github.com/Byron/gitoxide/blob/3c4e0b4997e86aa85d303da188f5eb55905bc1f2/gitoxide-core/src/pack/create.rs#L288-L291) can be used as guide on how to use the plumbing-level API for that.

I hope that helps.
